_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+ * This is the base class for every Core API endpoint Jetpack uses.
+ *
+ */
+class Jetpack_Core_API_Module_Activate_Endpoint
+	extends Jetpack_Core_API_XMLRPC_Consumer_Endpoint {
+
+	private $modules_requiring_public = array(
+		'sitemaps',
+		'photon',
+		'enhanced-distribution',
+		'sharedaddy',
+		'json-api',
+	);
+
+	public function process( $data ) {
+		if (
+			! in_array( $data['slug'], $this->modules_requiring_public )
+			|| $this->is_site_public()
+		) {
+			return Jetpack::activate_module( $data['slug'], false, false );
+		}
+		return new WP_Error(
+			'rest_cannot_publish',
+			__( 'This module requires your site to be set to publicly accessible.' ),
```

Comment:
Thanks for noticing! On escaping though, do you think it's necessary? The endpoint should do all the escaping when preparing the JSON it's going to send back to the client.